### PR TITLE
Mention Coq in the overview.

### PR DESCRIPTION
**Pull request by Zimmi48**
Since #2531, there is no longer a Coq section in the documentation, and since #2535 there is no longer a Coq sub-section in the "Other topics" section.  Coq's documentation is only available through its dedicated stanzas.  This means that it is much harder to discover that Dune supports not just OCaml and Reason, but also Coq.  This is an attempt to fix that, although it might be controversial, especially given that Coq's support is still very much a work-in-progress.

### CODE REVIEW

File: doc/overview.rst [hunk 1]
```diff
@@ -5,10 +5,11 @@ Overview
 Introduction
 ============
 
-Dune is a build system for OCaml and Reason. It is not intended as a
-completely generic build system that is able to build any given project
-in any language. On the contrary, it makes lots of choices in order to
-encourage a consistent development style.
+Dune is a build system for OCaml and Reason (with Coq support in
+progress).  It is not intended as a completely generic build system
```
Comment (nojb) [inline doc/overview.rst]: I would change the first sentence to say : 
```
Dune is a build system for OCaml (with support for Reason and Coq).
```
Am not sure it is necessary to mention that Coq support is in progress; the exact level of support is explained in the stanza documentation.

Comment (avsm) [reply doc/overview.rst]: Agreed with @nojb on the opening statement.

Comment (emillon) [reply doc/overview.rst]: And it'll age better :)

Comment (Zimmi48) [reply doc/overview.rst]: Thanks for the suggestions :grinning:

**Comment by nojb** [comment]
Thanks!

**Pull request merged**